test/System.IdentityModel.Tokens.Jwt.Tests/JwtSecurityTokenHandlerTests.cs

```diff
@@ -18,6 +18,7 @@
 
 namespace System.IdentityModel.Tokens.Jwt.Tests
 {
+    [CollectionDefinition("JwtSecurityTokenHandlerTests", DisableParallelization = true)]
```

Comment:
Why is this needed? Do tests fail running in parallel? Unit tests should be independent of each other.

Reply:
Yeap, AppContext is a static dict so, it gives me inconsistent behavior when setting the AppContext when tests run in parallel causing them to fail intermittently.

Reply:
Hmm, this should be inconsistent too then...

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/28e8784bf62126ea1071cff754e709b763d94cbd/test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenHandlerTests.cs#L4196

Reply:
What about moving the tests that use AppContext into a separate file, like JwtSecurityTokenHandlerTests.WithContextSwitchcs or something, and disable parallelization only there; so that these tests will run in parallel.

Reply:
I agree, They do cause a bit of an issue. I'll refactor this. Thanks Peter!

Reply:
I moved the test pertinent to this work to a separate file. Will create a task to potentially move the other tests on a separate PR if needed as they do not seem to encounter the same issue.